privatek8s.tf

```
@@ -169,6 +182,55 @@ resource "azurerm_kubernetes_cluster_node_pool" "infraciarm64" {
   tags = local.default_tags
 }
 
+# nodepool dedicated for the infra.ci.jenkins.io controller
+resource "azurerm_kubernetes_cluster_node_pool" "infracictrlarm64" {
+  name                  = "arm64infracictrl"
```

As per https://learn.microsoft.com/en-us/azure/aks/create-node-pools#limitations, the Linux node pool names are limited to 12 chars max. As @smerle33 reminded me, the Azure API does not validate the name during the "terraform plan" but only at the "terraform apply".

The `arm64infracictrl` is 16 characters which will fail once the PR is merged.

As such, I propose the following with 11 characters (room to add a suffix number if needed):

```suggestion
resource "azurerm_kubernetes_cluster_node_pool" "infraci_controller" {
  name                  = "infracictrl"
```

- Terraform resource identifier has no character limitation such as Azure API: we can use readable names
- Removing `arm64` from both Terraform Resource identifier and the node pool name: we only indicate the node pool role (e.g. "hosting infra.ci controller" or "hosting release.ci controller")
- Keeping the `ctrl` (see above)

WDYT?